akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/EventSourcedBehavior.scala

```diff
@@ -31,6 +31,10 @@ abstract class EventSourcedBehavior[Command, Event, State] private[akka] (
   }
 
   /**
+   * If using onPersistFailure the supervision is only around the event sourced behavior not any outer setup/withTimers
+   * block. If using restart any actions e.g. scheduling timers, can be done on the PreRestart signal or on the
```

Comment:
The `PreRestart` signal says the following:
> Lifecycle signal that is fired upon restart of the Actor before replacing the behavior with the fresh one (i.e. this signal is received within the behavior that failed).

Given that is the case, it seems impossible to use this signal to affect the *new* behavior. Or am I missing something?